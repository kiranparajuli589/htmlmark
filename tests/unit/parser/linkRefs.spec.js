import HtmlMark from "../../../lib/index.js"


describe("Link Reference", () => {
	const htmlMarkWithLinkRefsOn = new HtmlMark()
	const htmlMarkWithLinkRefsOff = new HtmlMark({ useLinkRefs: false })

	const lines = `[I'm an inline-style link](https://www.somewebsite.com)

[I'm an inline-style link with title](https://www.somewebsite.com "somewebsite's Homepage")

[I'm a reference-style link][Arbitrary case-insensitive reference text]

[I'm a relative reference to a repository file](../blob/master/LICENSE)

[You can use numbers for reference-style link definitions][1]

Or leave it empty and use the [link text itself]

Some text to show that the reference links can follow later.

[arbitrary case-insensitive reference text]: https://www.somewebsite.org
[1]: http://somewebsite.org "somewebsite's Homepage"
[link text itself]: http://www.somewebsite.com`

	it("should use the link references for link construction", () => {
		const html = htmlMarkWithLinkRefsOn.parse(lines)
		expect(html).toMatchSnapshot()
	})

	it("should not use the link references for link construction when disabled from config", () => {
		const html = htmlMarkWithLinkRefsOff.parse(lines)
		expect(html).toMatchSnapshot()
	})
})
