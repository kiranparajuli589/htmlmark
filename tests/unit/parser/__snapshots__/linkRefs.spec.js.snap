// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Reference should not use the link references for link construction when disabled from config 1`] = `
"<p><a href=\\"https://www.somewebsite.com\\">I&#39;m an inline-style link</a></p>
<p><a href=\\"https://www.somewebsite.com\\" title=\\"somewebsite's Homepage\\">I&#39;m an inline-style link with title</a></p>
<p>[I&#39;m a reference-style link][Arbitrary case-insensitive reference text]</p>
<p><a href=\\"../blob/master/LICENSE\\">I&#39;m a relative reference to a repository file</a></p>
<p>[You can use numbers for reference-style link definitions][1]</p>
<p>Or leave it empty and use the [link text itself]</p>
<p>Some text to show that the reference links can follow later.</p>
<p>[arbitrary case-insensitive reference text]: https://www.somewebsite.org [1]: http://somewebsite.org &quot;somewebsite&#39;s Homepage&quot; [link text itself]: http://www.somewebsite.com</p>
"
`;

exports[`Link Reference should use the link references for link construction 1`] = `
"<p><a href=\\"https://www.somewebsite.com\\">I&#39;m an inline-style link</a></p>
<p><a href=\\"https://www.somewebsite.com\\" title=\\"somewebsite's Homepage\\">I&#39;m an inline-style link with title</a></p>
<p>[I&#39;m a reference-style link][Arbitrary case-insensitive reference text]</p>
<p><a href=\\"../blob/master/LICENSE\\">I&#39;m a relative reference to a repository file</a></p>
<p><a href=\\"http://somewebsite.org\\" title=\\"somewebsite\\">You can use numbers for reference-style link definitions</a></p>
<p>Or leave it empty and use the <a href=\\"http://www.somewebsite.com\\">link text itself</a></p>
<p>Some text to show that the reference links can follow later.</p>
"
`;
